# Fitness Calories Prediction

## 1. Introduction

Predicting calories burned during exercise is a useful problem in health, fitness, and sports analytics.

## 2. Problem Statement

This is a supervised regression problem where we predict calories burned based on physiological and workout features.

In [3]:
import warnings
warnings.filterwarnings('ignore')

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from pathlib import Path

pd.set_option('display.max_columns', None)
sns.set_theme(style='whitegrid')

In [4]:
DATA_DIR = Path('../data/raw')

exercise_df = pd.read_csv(DATA_DIR / 'exercise_dataset.csv')
gym_df = pd.read_csv(DATA_DIR / 'gym_members_exercise_tracking.csv')

print(exercise_df.shape)
print(gym_df.shape)

(3864, 12)
(973, 15)


In [5]:
# ── 3. Data Inspection ──────────────────────────────────────────
print("=== GYM DATASET — first 5 rows ===")
display(gym_df.head())

print("\n=== EXERCISE DATASET — first 5 rows ===")
display(exercise_df.head())

print("\n=== MISSING VALUES ===")
print("Gym dataset:     ", gym_df.isnull().sum().sum(), "missing values")
print("Exercise dataset:", exercise_df.isnull().sum().sum(), "missing values")

print("\n=== BASIC STATISTICS — Gym dataset ===")
display(gym_df.describe().round(2))

=== GYM DATASET — first 5 rows ===


,Age,Gender,Weight (kg),Height (m),Max_BPM,Avg_BPM,Resting_BPM,Session_Duration (hours),Calories_Burned,Workout_Type,Fat_Percentage,Water_Intake (liters),Workout_Frequency (days/week),Experience_Level,BMI
0,56,Male,88.3,1.71,180,157,60,1.69,1313.0,Yoga,12.6,3.5,4,3,30.20
1,46,Female,74.9,1.53,179,151,66,1.30,883.0,HIIT,33.9,2.1,4,2,32.00
2,32,Female,68.1,1.66,167,122,54,1.11,677.0,Cardio,33.4,2.3,4,2,24.71
3,25,Male,53.2,1.70,190,164,56,0.59,532.0,Strength,28.8,2.1,3,1,18.41
4,38,Male,46.1,1.79,188,158,68,0.64,556.0,Strength,29.2,2.8,3,1,14.39



=== EXERCISE DATASET — first 5 rows ===


,ID,Exercise,Calories Burn,Dream Weight,Actual Weight,Age,Gender,Duration,Heart Rate,BMI,Weather Conditions,Exercise Intensity
0,1,Exercise 2,286.959851,91.892531,96.301115,45,Male,37,170,29.426275,Rainy,5
1,2,Exercise 7,343.453036,64.165097,61.104668,25,Male,43,142,21.286346,Rainy,5
2,3,Exercise 4,261.223465,70.846224,71.766724,20,Male,20,148,27.899592,Cloudy,4
3,4,Exercise 5,127.183858,79.477008,82.984456,33,Male,39,170,33.729552,Sunny,10
4,5,Exercise 10,416.318374,89.960226,85.643174,29,Female,34,118,23.286113,Cloudy,3



=== MISSING VALUES ===
Gym dataset:      0 missing values
Exercise dataset: 0 missing values

=== BASIC STATISTICS — Gym dataset ===


,Age,Weight (kg),Height (m),Max_BPM,Avg_BPM,Resting_BPM,Session_Duration (hours),Calories_Burned,Fat_Percentage,Water_Intake (liters),Workout_Frequency (days/week),Experience_Level,BMI
count,973.00,973.00,973.00,973.00,973.00,973.00,973.00,973.00,973.00,973.00,973.00,973.00,973.00
mean,38.68,73.85,1.72,179.88,143.77,62.22,1.26,905.42,24.98,2.63,3.32,1.81,24.91
std,12.18,21.21,0.13,11.53,14.35,7.33,0.34,272.64,6.26,0.60,0.91,0.74,6.66
min,18.00,40.00,1.50,160.00,120.00,50.00,0.50,303.00,10.00,1.50,2.00,1.00,12.32
25%,28.00,58.10,1.62,170.00,131.00,56.00,1.04,720.00,21.30,2.20,3.00,1.00,20.11
50%,40.00,70.00,1.71,180.00,143.00,62.00,1.26,893.00,26.20,2.60,3.00,2.00,24.16
75%,49.00,86.00,1.80,190.00,156.00,68.00,1.46,1076.00,29.30,3.10,4.00,2.00,28.56
max,59.00,129.90,2.00,199.00,169.00,74.00,2.00,1783.00,35.00,3.70,5.00,3.00,49.84


## 2. Problem Statement

### 2.1 Motivation
Physical inactivity is a leading risk factor for cardiovascular disease (WHO, 2022).
Predicting caloric expenditure can support personalized fitness recommendations
without expensive laboratory equipment.

### 2.2 Mathematical Formulation
Given feature matrix **X** ∈ ℝⁿˣᵈ, we learn function **f** such that:

    ŷ = f(X) ≈ y

where **y = Calories_Burned** is the continuous target variable.

We minimize Mean Squared Error:

    MSE = (1/n) · Σᵢ (yᵢ - ŷᵢ)²

### 2.3 Assumptions & Constraints
- Linear relationship assumed for the baseline model
- Dataset limited to gym members (selection bias possible)
- No dietary or sleep data available
- Sample size: 973 (gym) + 3864 (exercise dataset)

### 2.4 Related Work
| Approach | Advantage | Limitation |
|---|---|---|
| Linear Regression | Interpretable | Misses nonlinearity |
| Random Forest | High accuracy | Less interpretable |
| Neural Network | Best performance | Needs large data |